src/backend/utils/vkbasalt.py

```diff
@@ -0,0 +1,181 @@
+        if args.fxaa_subpixel_quality:
+            if 0 <= args.fxaa_subpixel_quality <= 1:
+                file.append(f"fxaaQualitySubpix = {args.fxaa_subpixel_quality}\n")
+            else:
+                stderr.write("Error: FXAA subpixel quality must be above 0 and below 1\n")
+                exit(1)
+
+        # --fxaa-edge-quality-threshold
+        if args.fxaa_quality_edge_threshold:
+            if 0 <= args.fxaa_quality_edge_threshold <= 1:
+                file.append(f"fxaaQualityEdgeThreshold = {args.fxaa_quality_edge_threshold}\n")
+            else:
+                stderr.write("Error: FXAA edge quality threshold must be above 0 and below 1\n")
+                exit(1)
+
+        # --fxaa-quality-edge-threshold-min
+        if args.fxaa_quality_edge_threshold_min:
+            if 0 <= args.fxaa_quality_edge_threshold_min <= 0.1:
+                file.append(f"fxaaQualityEdgeThresholdMin = {args.fxaa_quality_edge_threshold_min}\n")
+            else:
+                stderr.write("Error: FXAA edge quality threshold minimum must be above 0 and below 0.1\n")
+                exit(1)
+
+        # --smaa-edge-detection
+        if args.smaa_edge_detection:
+            file.append(f"smaaEdgeDetection = {args.smaa_edge_detection}\n")
+
+        # --smaa-threshold
+        if args.smaa_threshold:
+            if 0 <= args.smaa_threshold <= 0.5:
+                file.append(f"smaaThreshold = {args.smaa_threshold}\n")
+            else:
+                stderr.write("Error: SMAA threshold must be above 0 and below 0.5\n")
+                exit(1)
+
+        # --smaa-max-search-steps
+        if args.smaa_max_search_steps:
+            if 0 <= args.smaa_max_search_steps <= 112:
+                file.append(f"smaaMaxSearchSteps = {args.smaa_max_search_steps}\n")
+            else:
+                stderr.write("Error: SMAA max search steps must be above 0 and below 112\n")
+                exit(1)
+
+        # --smaa-max-search-steps-diagonal
+        if args.smaa_max_search_steps_diagonal:
+            if 0 <= args.smaa_max_search_steps_diagonal <= 20:
+                file.append(f"smaaMaxSearchStepsDiag = {args.smaa_max_search_steps_diagonal}\n")
+            else:
+                stderr.write("Error: SMAA max search steps diagonal must be above 0 and below 20\n")
+                exit(1)
+
+        # --smaa-corner-rounding
+        if args.smaa_corner_rounding:
+            if 0 <= args.smaa_corner_rounding <= 100:
+                file.append(f"smaaCornerRounding = {args.smaa_corner_rounding}\n")
+            else:
+                stderr.write("Error: SMAA corner rounding must be above 0 and below 100\n")
+                exit(1)
+
+        # --lut-file-path
+        if args.lut_file_path:
+            file.append(f"lutFile = {args.lut_file_path}\n")
+
+        # Output file
+        if args.output:
+            if path.isdir(args.output):
+                f = open(f"{args.output}/vkBasalt.conf", "w")
```

Comment:
Please always use `with open(..., ...) as f:` when doing IO, it takes into account a few things that might fail otherwise, like automatically closing the file when it goes out of scope or when there is a IO error (which is more important).